R/get-grades.R

```diff
@@ -20,18 +21,26 @@
 #' @importFrom tibble as_tibble
 #'
 #' @export
-get_grades <- function(gs, policy, verbose = FALSE) {
+get_grades <- function(gs, policy, verbose = FALSE, append = TRUE) {
   gs <- gs |>
     process_gs()
 
   policy <- policy |>
     process_policy(verbose = verbose) |>
     reconcile_policy_with_gs(gs = gs, verbose = verbose)
 
-  gs |>
+  grades <- gs |>
     apply_slip_days(policy = policy) |>
     calculate_grades(policy = policy) |>
     tibble::as_tibble()
+  
+  if (!append){
+    # id_cols calls all ID columns and all newly generated columns
+    # when it's called on a post-grading dataframe
+    grades <- grades[, get_id_cols(grades)]
```

Comment:
this approach makes sense to me. Could you remind me: why is `get_id_cols` is called that? Now we seem to differentiate between "assignments" (things that appear in the original GS file) and "categories". would `get_assignments()` or `get_assignment_cols()` be a sensible replacement?

I'm asking from the viewpoint of having come back to this package after some months of not working on it. It's so helpful to have clearly named functions :)